### ISSUE TITLE
Need nicer error message for checkbox with button role and missing aria-pressed

### ISSUE THREAD

**Issue by carmacleod** [IBMa/equal-access#337]
Steps:
- Go to: https://thepaciellogroup.github.io/aria-test-builder/dist/testcases/input-checkbox-test.html
- Run Accessibility Checker

Violation: "The WAI-ARIA role or attribute 'button' is not valid for the element `<input>`"
Actually, role="button" is allowed on a checkbox input if it has the required aria-pressed attribute.

The trouble with the current error message is twofold:
- it's not specific to "checkbox" input element
- it's too harsh - it should say something like, "when using a WAI-ARIA role on an element, the required attributes for that role must be defined." ... ideally, it could very specifically say something like, "The WAI-ARIA role or attribute 'button' is not valid for the element `<input type="checkbox">` unless the required aria-pressed attribute is also present"

![image](https://user-images.githubusercontent.com/3331913/110388315-e09b6d00-8030-11eb-9cb5-832fb0be679a.png)

**Comment by MHoov** [comment]
As of June 1
- [x] general rule description (SME & Dev)
- [x] testcases written (Dev)
- [x] rule developed (Dev)
- [ ] message written (SME)
- [x] help written (SME)

**Comment by philljenkins** [comment]
Reviewing the issue and writing help & messages changes

**Comment by tombrunet** [comment]
I think this is actually a logic issue:
1) The confusing part of the token is the element name. This is something we should probably look at more generally. The message string is `<{1}>` which is filled with the element name. We've seen a couple of times that `<input>` isn't really specific enough. If the element has a `type` attribute, we should show `<input type=xxxxx>`.
2) Need reason codes to distinguish between role not valid at all and role not valid because attribute missing.

**Comment by tombrunet** [comment]
Going to remove SME because I think we need to do some rule work first.

**Comment by philljenkins** [comment]
Without changes in logic appears as a false positive. I don’t think there’s much to be done on the message / help side until we get the logic straightened out.

**Comment by MHoov** [comment]
@philljenkins is still assigned, does this need to be assigned to @shunguoy ?

**Comment by tombrunet** [comment]
@MHoov Since the work changed and this was put back into the backlog, should the sprints be removed?

**Comment by philljenkins** [comment]
I unassigned myself so it can be assigned to a dev

**Comment by marcjohlic** [comment]
I removed all of the metadata for sprint, epic, and release.   This MAY still increment sprints unless ZH has fixed that bug.

**Comment by marcjohlic** [comment]
Shunguo suggests waiting until we complete the current Rules deployment that we're working on.  Fixes in that deployment may fix this issue.

**Comment by shunguoy** [comment]
open a new issue to distinguish "aria_semantics" (role not valid) and "aria_rule_redundant", and close the current one
